src/controller.rs

```diff
@@ -139,6 +141,13 @@ async fn launch_job(
     let ns = &app_instance.namespace().ok_or(Error::NamespaceRequired)?;
     let job_name = format!("kubit-apply-{}", app_instance.name_any());
 
+    let tmp = tempfile::Builder::new().suffix(".json").tempfile()?;
+    let (mut file, path) = tmp.keep()?;
+    serde_json::to_writer(&mut file, &app_instance).map_err(Error::RenderOverlay)?;
```

Comment:
I don't think this is actually correct, but I took it from the `emit_script` function as this seems to be how an overlay is applied.

It seems to give the correct-ish results for now, but the logic seems odd to me. I.e. pass in an app_instance CR already, but use that as an overlay on the inbound resource too.

Reply:
It occurred to me that this will be needed, but the results of this will be placed into a `ConfigMap` so that the created `Job` could access it from a mount.

Reply:
**This has since changed, so I'll resolve this and add more comments**